monitoring/monit.py

```diff
@@ -86,8 +88,35 @@ def run_command(command):
         module.run_command('%s %s %s' % (MONIT, command, name), check_rc=True)
         return status()
 
-    process_status = status()
-    present = process_status != ''
+    def wait_for_monit_to_stop_pending(sleep_time=1):
```

Comment:
I think you need more than 10s, some services can take minutes to startup or shutdown. I'd suggest a timeout of say 5m rather than a retry limit, and have say 10s between attempts.